Fix month dropdown

KevinMann13 opened the pull request:
Hey Jesse, I don't really understand why you were looping over months and then looping over monthstuff. I went ahead and removed the second loop and just pulled from the name on the initial loop. That then puts the right data in to the option which is then set to the ng-model which is then printed out by the update function. Let me know if I completely misunderstood what was supposed to be happening.

Commit:
Fix month dropdown

jrountree commented:
Thanks for fixing that.  I spent a month reading forms and watching videos and experimenting with it, and could never get it to read past the first array in the json file, no idea why.  So i found that Work around, which may have cause the current error.  I check out what you did tomorrow and commit it to the mast branch if it works.  I really appreciate it man.  It’s like a wall I can’t get by and now, hopefully, I can move forward.

Sent from my iPhone

> On Dec 19, 2017, at 7:40 AM, Kevin Mann <notifications@github.com> wrote:
> 
> Hey Jesse, I don't really understand why you were looping over months and then looping over monthstuff. I went ahead and removed the second loop and just pulled from the name on the initial loop. That then puts the right data in to the option which is then set to the ng-model which is then printed out by the update function. Let me know if I completely misunderstood what was supposed to be happening.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/jrountree/Dashboard-Using-Angularjs-1/pull/2
> 
> Commit Summary
> 
> Fix month dropdown
> File Changes
> 
> M index.html (8)
> Patch Links:
> 
> https://github.com/jrountree/Dashboard-Using-Angularjs-1/pull/2.patch
> https://github.com/jrountree/Dashboard-Using-Angularjs-1/pull/2.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

jrountree commented:
Thank you Kevin!  I looped over months and then looped over monthstuff because that was the only was I could get the ng-repeat to access the array inside of the object in the json file....

json looks like this....

[
{ "monthstuff": [

	{	"name": "January",
		"number": 1,
		"weeks": [
			{"id":1, "info": 8},
			{"id":2, "info": 40},
			{"id":3, "info": 75},
			{"id":4, "info": 65},
			{"id":5, "info": 90},
			{"id":6, "info": 8}
			]
				
	},
....etc.  

I tried the following html:
option ng-repeat="month in monthdata.monthstuff" {{month.name}}   /option  

That did not work.  I found that in order to access the information inside of the monthstuff array, i had to first contain everything inside of a loop that accessed the array.  THEN I can have the following html:

 option ng-repeat="monthlist in monthdropdown.monthstuff" value="{{monthlist.name}}"{{monthlist.name}} /option

...where "monthdropdown" is the container loop, and now I can access the information with it.  However, this means that the ng-change is unable to grab that information because it is looking for it inside of the container loop and not the specific array in the monthlist, although that is just my best guess.  So my question is, how do I avoid having to use a container loop, like the way you have it above, but access the information inside of the array?  Because as you have it now, it does not pull the month "name" and my drop down menu is empty.  Thanks!

Commit:
Getting the list of cups by week working

KevinMann13 commented:
You must not have committed those changes to the JSON, because I am seeing JSON that looked like this:

    {		  	
        {
  	    "month": "January",
            "Week1": 8,
 	    "Week2": 40,		
 	    "Week3": 75,		
 	    "Week4": 65,		
 	    "Week5": 90,		
 	    "Week6": 8		
  	}
    }

I would suggest that you do a pull from my code into a seperate directory, so you aren't looking at the uncommitted changes on your side at the same time.

    git clone -b km https://github.com/KevinMann13/Dashboard-Using-Angularjs-1.git

I went ahead and created another commit to get the list of weeks/cups for the month working. It is now working where if I select a month, I see the weeks/cups repeat down the page. I modified the JSON to be as simple as possible.

    {		  	
        {
  	    "month": "January",
            "weeks": [8,40, 75,65, 90, 8]
  	}
    }

The major change that I made is that I incorporated ng-value instead of just value which allows me to store an object in it. So when you select "January", the whole "January" object is stored with it with the weeks and all. Then further down the page I can loop over the weeks in that selected object to get their values.

Another thing is I am using $index as the counter. It is a built in part of ng-repeat that gives me the index of the current value. I am then adding 1 to it so it doesn't start at 0 like a normal index.

Are you free some time today? We can get on a call so that I can get you past these blockers. I think once you get this concept, it will really help you moving forward.
